source/postprocess/melt_statistics.cc

```diff
@@ -0,0 +1,183 @@
+                                  in.pressure);
+            fe_values[this->introspection().extractors.velocities]
+            .get_function_values (this->get_solution(),
+                                  in.velocity);
+            fe_values[this->introspection().extractors.pressure]
+            .get_function_gradients (this->get_solution(),
+                                     in.pressure_gradient);
+            for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+              fe_values[this->introspection().extractors.compositional_fields[c]]
+              .get_function_values(this->get_solution(),
+                                   composition_values[c]);
+            for (unsigned int i=0; i<fe_values.n_quadrature_points; ++i)
+              {
+                for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+                  in.composition[i][c] = composition_values[c][i];
+              }
+
+            fe_values[this->introspection().extractors.velocities].get_function_symmetric_gradients (this->get_solution(),
+                in.strain_rate);
+            in.position = fe_values.get_quadrature_points();
+            in.cell = &cell;
+
+            // we can only postprocess melt fractions if the material model that is used
+            // in the simulation has implemented them
+            // otherwise, set them to zero
+            std::vector<double> melt_fractions(n_q_points, 0.0);
+            if (const MaterialModel::MeltFractionModel<dim> *
+                melt_material_model = dynamic_cast <const MaterialModel::MeltFractionModel<dim>*> (&this->get_material_model()))
+              melt_material_model->melt_fractions(in, melt_fractions);
+
+            for (unsigned int q=0; q<n_q_points; ++q)
+              {
+                local_melt_integral += melt_fractions[q] * fe_values.JxW(q);
+                local_min_melt       = std::min(local_min_melt, melt_fractions[q]);
+                local_max_melt       = std::max(local_max_melt, melt_fractions[q]);
+              }
+
+          }
+
+      const double global_melt_integral
+        = Utilities::MPI::sum (local_melt_integral, this->get_mpi_communicator());
+      double global_min_melt = 0;
+      double global_max_melt = 0;
+
+      // now do the reductions that are
+      // min/max operations. do them in
+      // one communication by multiplying
+      // one value by -1
+      {
+        double local_values[2] = { -local_min_melt, local_max_melt };
+        double global_values[2];
+
+        Utilities::MPI::max (local_values, this->get_mpi_communicator(), global_values);
+
+        global_min_melt = -global_values[0];
+        global_max_melt = global_values[1];
+      }
+
+
+      // finally produce something for the statistics file
+      statistics.add_value ("Minimal melt fraction",
+                            global_min_melt);
+      statistics.add_value ("Total melt fraction",
+                            global_melt_integral);
```

Comment:
I think this is a bit unclear: you call the column "fraction", but you're really outputting the total *amount*. Maybe rephrase? Or indeed compute the fraction? Same below in what's returned by this function.